### PR TITLE
ui: Add Experimental "ClusterViz" page

### PR DESCRIPTION
**Pull request by mrtracy**
Adds an experimental page to the admin UI, accessible by navigating to the
"/clusterViz" URL manually. This page is not currently reachable through
link-based navigation.

This page hooks up a visualization (and associated simulation) developed by
Spencer to showcase the behavior of CockroachDB; the intent is that this
visualization will be the starting point for more advanced cluster
visualizations using real data.

The javascript code being checked in is currently not subject to any linters,
and is a bit rough around the edges; it has also been rapidly converted into
commonjs so that it can be loaded into our UI. For the time being, it should not
be scrutinized through our normal code review process; it is subject to rapid
change and much of it will likely be discarded in the short term.

screen shot 2017-02-03 at 5 44 23 pm

---
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/13401)

### CODE REVIEW

**Comment by petermattis** [comment]
Very exciting!

File: pkg/ui/app/containers/clusterViz.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import * as React from "react";
+import {runVisualization, viewWidth, viewHeight} from "../js/sim/index";
+
+/**
+ * Renders the layout of the ClusterViz page. "ClusterViz" is an experimental
+ * visual map of the cluster, expressed as a directive
```
Comment (spencerkimball) [inline pkg/ui/app/containers/clusterViz.tsx]: Is this comment complete? Or does it just need end punctuation? I don't know what a directive is in this context...

File: pkg/ui/app/js/sim/index.html [hunk 2]
```diff
@@ -0,0 +1,50 @@
+<!DOCTYPE html>
```
Comment (spencerkimball) [inline pkg/ui/app/js/sim/index.html]: Does this file need to be in here?

File: pkg/ui/app/js/sim/index.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
+var d3 = require("d3");
+var Model = require("./model.js");
+var Visualization = require("./visualization.js");
+var Datacenter = require("./datacenter.js");
+var AggV1 = require("./aggv1.js");
+
+function initModel(el, model) {
+  var viewHeight = Visualization.viewHeight;
```
Comment (spencerkimball) [inline pkg/ui/app/js/sim/index.js]: What's the point of these two vars?

File: pkg/ui/app/js/sim/raft_quiescence.html [hunk 4]
```diff
@@ -0,0 +1,63 @@
+<!DOCTYPE html>
```
Comment (spencerkimball) [inline pkg/ui/app/js/sim/raft_quiescence.html]: Drop this file and also the .png files.

File: pkg/ui/app/js/sim/rebalance_thrashing.html [hunk 5]
```diff
@@ -0,0 +1,61 @@
+<!DOCTYPE html>
```
Comment (spencerkimball) [inline pkg/ui/app/js/sim/rebalance_thrashing.html]: Drop this file.

File: pkg/ui/app/js/sim/single_node.html [hunk 6]
```diff
@@ -0,0 +1,50 @@
+<!DOCTYPE html>
```
Comment (spencerkimball) [inline pkg/ui/app/js/sim/single_node.html]: Drop this file.

**Review by spencerkimball — APPROVED** [review]

File: pkg/ui/app/js/sim/index.js (hunk 3 above)
Comment (mrtracy) [inline pkg/ui/app/js/sim/index.js]: Need to set the width and the height of the div in which the visualization is being rendered..

File: pkg/ui/app/containers/clusterViz.tsx (hunk 1 above)
Comment (mrtracy) [inline pkg/ui/app/containers/clusterViz.tsx]: Trailed off in the middle of a thought; I was probably trying to write "directed graph", decided that it wasn't a directed graph, left to figure out what it actually was, then never came back to this comment.

File: pkg/ui/app/js/sim/index.js (hunk 3 above)
Comment (mrtracy) [inline pkg/ui/app/js/sim/index.js]: Oh, was looking at the wrong file. I didn't need these.

File: pkg/ui/app/js/sim/index.html (hunk 2 above)
Comment (mrtracy) [inline pkg/ui/app/js/sim/index.html]: The entire file can be deleted.

**Comment by mrtracy** [comment]
Comments addressed.

**Comment by spencerkimball** [comment]
LGTM

**Comment by spencerkimball** [comment]
@mrtracy go ahead and merge this.

**Pull request merged**